UnityProject/Assets/Scripts/Core/RootSillys/SingletonScriptableObject.cs

```diff
@@ -0,0 +1,60 @@
+using System.Diagnostics;
+using System.Linq;
+using UnityEngine;
+using Debug = System.Diagnostics.Debug;
+
+namespace ScriptableObjects
+{
+	/// <summary>
+	/// Abstract class for making reload-proof singletons out of ScriptableObjects
+	/// Returns the asset created on the editor, or null if there is none
+	/// Based on https://www.youtube.com/watch?v=VBA1QCoEAX4
+	/// </summary>
+	/// <typeparam name="T">Singleton type</typeparam>
+
+	public abstract class BadSingletonScriptableObject<T> : ScriptableObject where T : ScriptableObject
```

Comment:
why is it called bad?

Reply:
this is due to lighting system Being in its own assembly and being annoying, so have to copy this to make the common materials SO singleton worrk ,it's bad because it doesn't reference the  statiic manager with all the SO preloaded